src/gp_doctest.cpp

```diff
@@ -1989,3 +1987,100 @@ TEST_CASE("Top-Pruning: Likelihoods") {
       fasta_path_six, newick_path_six_simple, false, false);
   CHECK_MESSAGE(test_3, "Six Taxa Multi Tree failed.");
 }
+
+// Runs an instance of TPEngine for two DAGs: DAG_1, a simple DAG, and DAG_2, a DAG
+// formed from DAG_1 plus all of its adjacent NNIs. Both DAGs PVs are populated and
+// their edge TP likelihoods are computed.  Then DAG_1's adjacent proposed NNI
+// likelihoods are computed using only its own PVs. Then we compare the results of
+// the proposed NNIs from DAG_1 with the known likelihoods of DAG_2.
```

Comment:
Can you add a sentence saying what the comparison should show? The setup sounds reasonable, but I don't quite follow the last part.

Reply:
Here's an update.  Is this clearer?

```
// Runs an instance of TPEngine for two DAGs: DAG_1, a simple DAG, and DAG_2, a DAG
// formed from DAG_1 plus all of its adjacent NNIs. Both DAGs PVs are populated and
// their edge TP likelihoods are computed.  Then DAG_1's adjacent proposed NNI
// likelihoods are computed using only PVs from the pre-NNI already contained in DAG_1.
// Finally, we compare the results of the proposed NNIs from DAG_1 with the known
// likelihoods of the actual NNIs already contained in DAG_2. This verifies we generate
// the same result from adding NNIs to the DAG and updating as we do from using the
// pre-NNI computation.
```

Reply:
Perfect.